contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -1720,3 +1726,45 @@ preserve_view_constraints_from_base_table(ColumnDef  *col, Oid tableOid, AttrNum
 		}
 	}
 }
+
+/*
+ * exact_numeric_digit_count() -
+ * 	Calculate exact number of digits of any numeric data
+ */
+int
+pltsql_exact_numeric_digit_count(int weight, int dscale, int first_block, int numeric_base)
+{
+	int partially_filled_numeric_block = 0;
+	int total_digit_count = 0;
+
+	total_digit_count = (dscale == 0) ? (weight * numeric_base) :
+						((weight + 1) * numeric_base);
+	/*
+	 * calculating exact #digits in the first partially filled numeric block, if any)
+	 * Ex. - in 12345.12345 var is of type struct NumericVar; first_block = var->digits[0]= 1,
+	 * var->digits[1] = 2345, var->digits[2] = 1234,
+	 * var->digits[3] = 5000; numeric_base = 4, var->ndigits = #numeric blocks i.e., 4,
+	 * var->weight = 2, var->dscale = 5
```

Comment:
if take this example : total_digit_count = (2 + 1) * 4 = 12
+1 for first block and +1 for dscale leads to final 14 digits, can you explain how is that correct?

Reply:
Breakpoint 2, numeric_sum (fcinfo=<optimized out>) at numeric.c:6010
6010            if ((*detect_numeric_overflow_hook)(sumX_var.weight, sumX_var.dscale, sumX_var.digits[0], DEC_DIGITS))
(gdb) p sumX_var
$1 = {ndigits = 4, weight = 1, sign = 0, dscale = 5, buf = 0x7f1e16330178, digits = 0x7f1e1633017c}

weight = 1; I will rectify my comment in code

Reply:
similar calculation of fully filled block digits from numeric_base and weight can be found in numeric() in engine; have added extra calculation for partially filled 1st block and some checks whether last block was partially filled here only.